File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Response/InvoiceItemResponse.java
package ut.edu.be_quanlytro.Dto.Response;

import lombok.Builder;
import lombok.Data;
import java.math.BigDecimal;

@Data
@Builder
public class InvoiceItemResponse {
    private String serviceName;
    private Integer oldIndex;    // Sẽ null nếu là Wifi/Rác
    private Integer newIndex;    // Sẽ null nếu là Wifi/Rác
    private Integer quantity;    // Số ký điện, khối nước hoặc số đầu người
    private BigDecimal price;     // Đơn giá
    private BigDecimal totalAmount; // Thành tiền dòng này
}

File: Flutter_QuanLyTro/lib/data/models/response/invoice_item_response.dart
class InvoiceItemResponse {
  final String serviceName;
  final int? oldIndex;
  final int? newIndex;
  final int quantity;
  final double price;
  final double totalAmount;

  InvoiceItemResponse({
    required this.serviceName,
    this.oldIndex,
    this.newIndex,
    required this.quantity,
    required this.price,
    required this.totalAmount,
  });

  factory InvoiceItemResponse.fromJson(Map<String, dynamic> json) {
    return InvoiceItemResponse(
      serviceName: json['serviceName'] ?? '',
      oldIndex: json['oldIndex'],
      newIndex: json['newIndex'],
      quantity: json['quantity'] ?? 0,
      price: (json['price'] as num?)?.toDouble() ?? 0.0,
      totalAmount: (json['totalAmount'] as num?)?.toDouble() ?? 0.0,
    );
  }
}